pkg/agent/agentserver/server.go

```diff
@@ -61,17 +67,30 @@ type ProxyServer struct {
 	// connID track
 	Frontends   map[int64]*ProxyClientConnection
 	PendingDial map[int64]*ProxyClientConnection
+
+	// agent authentication
+	AgentAuthenticationOptions *AgentAuthenticationOptions
+}
+
+// AgentAuthenticationOptions contains list of parameters required for agent authentication
```

Comment:
Can you rename it to AgentTokenAuthenticationOptions and update the comment, since we still support the cert based authentication?

Reply:
done